Add usermanagement to resty linchpin #42(description of routes added to issue)

mansikulkarni96 opened the pull request.

Commit:
Added initial code for user management includes, create, delete, login, verify token functionality

mansikulkarni96 committed:
not required file

Commit:
Added flag for admin user, added conditions for checking admin access

Commit:
Merge branch 'usr_manage' of https://github.com/mansikulkarni96/restylinchpin into usr_manage

merge

Commit:
Added response messages, admin user creation, delete, reset API key routes

Commit:
Added put API route for updating user info, added user access rights to do operation on owner user accounts

Commit:
Added authenctication for all workspace options

mansikulkarni96 committed:
Merge pull request #3 from CentOS-PaaS-SIG/develop

getting upstream changes

Commit:
Resolved conflicts

Commit:
Revert "Resolved conflicts"

This reverts commit 31e6cd4e7c12ce6c84dbcf6a65c572184afe069d.

Commit:
Accomodated code to new folder structure

Commit:
Added exception handling, moved functions to utils

Commit:
Edited doc strings, return types, response_messages

Commit:
Added basic auth for reset_api_key, modified routes for uniqueness

Commit:
Modified indentations, path for reset API key

Commit:
Removed unnecessary flags, fixed indentations

Commit:
Fixed flake8 errors

samvarankashyap commented on app/config.yml:
```diff
@@ -3,11 +3,17 @@
 logger_file_name: restylinchpin.log
-# path to tinydb source file
+#path to users tinydb source file
+users_db_path: users.json
```
@mansikulkarni96 
Do we need to have another json file, 
can use the same db.json and create another record for users ?? 
I don't see any advantages on using another db for the same application

samvarankashyap commented on app/__init__.py:
```diff
@@ -7,35 +7,40 @@
     with open(APP_DIR + '/config.yml', 'r') as f:
         config = yaml.load(f)
-except Exception as e:
+except Exception as x:
```
@mansikulkarni96 why x instead of e?

mansikulkarni96 replied:
changed it because it was being used globally and shadowing other usages, I could use some other variable related to exception.

samvarankashyap reviewed (changes requested):
Please see the comments
further, could you add workflow documentation to README
on how to create users and reset API keys and use API keys using this application
Thanks

Commit:
Created tables for workspace and users,combined usersdb with db.json

removed unnecessary repeated commit
Created tables for workspace and users,combined usersdb with db.json

Commit:
Fixed flake8 errors

mansikulkarni96 committed:
Updated readme for user management actions

updates to readme

fixed build failure

added other remaining options

removing unwanted commits

samvarankashyap reviewed (approved):
LGTM
